projects/plugins/social/src/js/components/admin-page/index.jsx

```diff
@@ -64,14 +66,50 @@ const ConnectionItems = () => {
 		);
 	} );
 };
+
+const ModuleToggle = () => {
+	const updateOptions = useDispatch( STORE_ID ).updateJetpackSettings;
+	const isModuleEnabled = useSelect( select => select( STORE_ID ).isModuleEnabled() );
+	const isUpdating = useSelect( select => select( STORE_ID ).isUpdatingJetpackSettings() );
+
+	const toggleModule = useCallback( () => {
+		const newOption = {
+			publicize_active: ! isModuleEnabled,
+		};
+		updateOptions( newOption );
+	}, [ isModuleEnabled, updateOptions ] );
+
+	const label = isModuleEnabled
+		? __( 'Jetpack Social is active', 'jetpack-social' )
+		: __( 'Jetpack Social is inactive', 'jetpack-social' );
+
+	return (
+		<ToggleControl
+			label={ __( 'Activate Jetpack Social', 'jetpack-social' ) }
+			help={ isUpdating ? __( 'Updating…', 'jetpack-social' ) : label }
+			disabled={ isUpdating }
+			checked={ isModuleEnabled }
+			onChange={ toggleModule }
+		/>
+	);
+};
+
 const Admin = () => {
+	useEffect( () => {
+		const apiRootUrl = syncSelect( STORE_ID ).getAPIRootUrl();
+		const apiNonce = syncSelect( STORE_ID ).getAPINonce();
```

Comment:
Good point, I'll switch it to a single call. `syncSelect` is what it's called in the Search plugin (what I partly based this PR on) and I believe renaming it to just `select` would cause naming conflicts with all the `select` parameters in the `useSelect` calls in the file.

Reply:
Switched in [a5fde97](https://github.com/Automattic/jetpack/pull/24203/commits/a5fde97a4249cbcce493994c81e26427f8d29ed0)

Reply:
Thanks! `syncSelect` doesn't seem like the best name, but it'll do for now 😄